core/src/main/java/org/mobilitydata/gtfsvalidator/notice/NoticeContainer.java

```diff
@@ -152,4 +173,195 @@ public String exportSystemErrors(boolean isPretty) {
+      parametersAsJsonArray.add(parameterDetails);
+    });
+    if (!hasSeverityLevelParameter(constructor)) {
+      JsonObject severityDetails = new JsonObject();
+      severityDetails.addProperty(JSON_KEY_NAME, CLASS_SIMPLE_NAME_SEVERITY_LEVEL);
+      severityDetails.addProperty(JSON_KEY_TYPE, BLOB_DATA_TYPE);
+      parametersAsJsonArray.add(severityDetails);
+    }
+    return parametersAsJsonArray;
+  }
+
+  /**
+   * Return true if a notice constructor includes a parameter of type {@code SeverityLevel},
+   * otherwise returns false.
+   *
+   * @param constructor the constructor to analyze
+   * @return true if a notice constructor includes a parameter of type {@code SeverityLevel},
+   * otherwise returns false.
+   */
+  private static boolean hasSeverityLevelParameter(Constructor<?> constructor) {
+    for (Parameter parameter : constructor.getParameters()) {
+      if (parameter.getType().getSimpleName().equals(CLASS_SIMPLE_NAME_SEVERITY_LEVEL)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Maps Java data type to Oracle's JDBC data types. Said types definitions can be found at the
+   * following url: https://docs.oracle.com/cd/E19830-01/819-4721/beajw/index.html.
+   *
+   * @param parameter the parameter whose JDBC data type has to be inferred
+   * @return the JDBC data type that is used in mapping Java data fields to SQL types
+   */
+  private static String mapDataType(Parameter parameter) {
+    ImmutableMap.Builder<Class<?>, String> builder = ImmutableMap.builder();
+    builder.put(int.class, INTEGER_DATA_TYPE);
+    builder.put(GtfsColor.class, VARCHAR_DATA_TYPE);
+    builder.put(String.class, VARCHAR_DATA_TYPE);
+    builder.put(long.class, BIGINT_DATA_TYPE);
+    builder.put(SeverityLevel.class, BLOB_DATA_TYPE);
+    builder.put(GtfsDate.class, BLOB_DATA_TYPE);
+    builder.put(GtfsTime.class, BLOB_DATA_TYPE);
+    builder.put(Object.class, BLOB_DATA_TYPE);
+    ImmutableMap<Class<?>, String> typeMap = builder.build();
+    return JDBCType.valueOf(
+        typeMap.getOrDefault(
+            parameter.getType(),
+            parameter.getType().getSimpleName().toUpperCase()))
+        .getName();
+  }
+
+  /**
+   * Returns the annotated constructor of a {@code ValidationNotice} subclass.
+   * Throws {@code AnnotationFormatError} if the {@code ValidationNotice} subclass does not define a
+   * constructor for schema export i.e. no constructor for the given class ses {@code NoticeExport}
+   * annotation.
+   *
+   * @param clazz the class to extract the annotated constructor from
+   * @return the annotated constructor of a {@code ValidationNotice} subclass.
+   * Throws {@code AnnotationFormatError} if the {@code ValidationNotice} subclass does not define a
+   * constructor for schema export i.e. no constructor for the given class uses {@code NoticeExport}
+   * annotation.
+   */
+  private static Constructor<?> getAnnotatedConstructor(Class<?> clazz) {
```

Comment:
Please support more than one constructor with `@SchemaExport` annotation.

Reply:
In this case, how would you make the difference between the two constructors for the same notice in the JSON schema?

Reply:
Will push the changes on this branch after clarifying this last point :)

Reply:
All params from all constructors annotated with `@SchemaExport` should be aggregated. Consider:

```
@SchemaExport
MyCoolNotice(long csvRowNumber, String routeId) {}
@SchemaExport
MyCoolNotice(long csvRowNumber, String tripId) {}
```